[Bug] [master] NPE exception when workflow fault tolerance start

### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

[ERROR] 2023-08-07 16:32:11.287 +0800 org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnable:[737] - [WorkflowInstance-0][TaskInstance-0] - Start workflow error
java.lang.NullPointerException: null
	at org.apache.dolphinscheduler.service.expand.CuringParamsServiceImpl.preBuildBusinessParams(CuringParamsServiceImpl.java:272)
	at org.apache.dolphinscheduler.server.master.runner.execute.TaskExecutionContextFactory.createTaskExecutionContext(TaskExecutionContextFactory.java:124)
	at org.apache.dolphinscheduler.server.master.runner.execute.DefaultTaskExecuteRunnableFactory.createTaskExecuteRunnable(DefaultTaskExecuteRunnableFactory.java:53)
	at org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnable.tryToTakeOverTaskInstance(WorkflowExecuteRunnable.java:1410)
	at org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnable.submitPostNode(WorkflowExecuteRunnable.java:1298)
	at org.apache.dolphinscheduler.server.master.runner.WorkflowExecuteRunnable.startWorkflow(WorkflowExecuteRunnable.java:731)
	at org.apache.dolphinscheduler.server.master.runner.IWorkflowExecuteRunnable.call(IWorkflowExecuteRunnable.java:27)




### What you expected to happen

refer What happened

### How to reproduce

When one master node hangs and another node is fault-tolerant

### Anything else

![image](https://github.com/apache/dolphinscheduler/assets/20450037/2eae1109-ee6a-4821-82b3-fdcfb9dd5cc8)

The null pointer is because 'processInstance' is null, which comes from the cached 'taskInstance' in the 'taskInstanceMap'. So it's necessary to ensure that the 'processInstance' has been placed in the 'taskInstance' before caching. And I don't think this is a good design with other attributes in DO.

### Version

dev

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java
+++ b/dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java
@@ -849,6 +849,7 @@ public class WorkflowExecuteRunnable implements IWorkflowExecuteRunnable {
                         }
                     }
 
+                    processService.packageTaskInstance(task, workflowInstance);
                     validTaskMap.put(task.getTaskCode(), task.getId());
                     taskInstanceMap.put(task.getId(), task);
                     taskCodeInstanceMap.put(task.getTaskCode(), task);

